scanner/decisions/0009-scannerv4-read-manual-vulnerabilities.md

```diff
@@ -0,0 +1,69 @@
+# 0009 - Improve Scanner V4 manually added vulnerabilities updating process
+
+- **Author(s):** Yi Li <yli3@redhat.com>
+- **Created:** [2024-04-30]
+
+## Context
+
+Currently, Scanner v4 includes a feature for manually updating vulnerabilities to add and update urgent vulnerability data. 
+This allows clients to access the most recent vulnerability information before it becomes officially available in some data sources. 
+However, this manual update process has a major flaw: the vulnerability data is integrated with the codebase, which means clients cannot access the updated data unless they install a patch release that includes these vulnerabilities. 
+Therefore, it is essential to improve this process to ensure that manually inserted vulnerabilities are not dependent on any ACS release cycle. This improvement would enable all versions starting from ACS 4.4 to retrieve manually updated vulnerabilities without any issues.
+
+## Decision
+
+Introduce a JSON file at /scanner/updater/manual/vulns.json (The format of the json will be listed below), and update the corresponding manual vulnerability updater located in scanner/updater/manual/manual.go. 
+This updater will be utilized within the existing vulnerability updating GitHub Actions workflow. 
+The new manual vulnerability updater will retrieve and parse the manually inserted data from https://github.com/stackrox/stackrox/blob/master/scanner/updater/manual/vulns.json. 
```

Comment:
Do you mean pulling directly from github?

Reply:
Github as a dependency is fine; we already depend on it to generate the bundles anyway. I  am not sure about the URL (I think you would want to use <https://raw.githubusercontent.com>.

FYI, another option might be having a step in the vuln workflow that generates "vulns.json" (as a new command in the `updater`) and stores it using `actions/upload-artifact@v4`. Then every matrix job could consume that. This would allow us to continue using Go structs to define the vulnerabilities; it would be interesting if we had to perform some complex logic to generate `vulns.json` (like iterating over data, mapping or filtering existing data, etc.). But that's not the case, AFAIK. Just keeping a JSON in the repo seems the simplest solution.